code/modules/mob/living/basic/clown/clown.dm

```diff
@@ -8,191 +8,113 @@
-	turns_per_move = 5
 	response_disarm_continuous = "gently pushes aside"
 	response_disarm_simple = "gently push aside"
 	response_harm_continuous = "robusts"
 	response_harm_simple = "robust"
-	speak = list("HONK", "Honk!", "Welcome to clown planet!")
-	emote_see = list("honks", "squeaks")
-	speak_chance = 1
 	combat_mode = TRUE
 	maxHealth = 75
 	health = 75
-	speed = 1
-	harm_intent_damage = 8
 	melee_damage_lower = 10
 	melee_damage_upper = 10
 	attack_sound = 'sound/items/bikehorn.ogg'
-	obj_damage = 0
+	attacked_sound = 'sound/items/bikehorn.ogg'
 	environment_smash = ENVIRONMENT_SMASH_NONE
-	del_on_death = 1
-	loot = list(/obj/effect/mob_spawn/corpse/human/clown)
+	basic_mob_flags = DEL_ON_DEATH
 	initial_language_holder = /datum/language_holder/clown
-	atmos_requirements = list("min_oxy" = 5, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 1, "min_co2" = 0, "max_co2" = 5, "min_n2" = 0, "max_n2" = 0)
-	minbodytemp = 270
-	maxbodytemp = 370
+	habitable_atmos = list("min_oxy" = 5, "max_oxy" = 0, "min_plas" = 0, "max_plas" = 1, "min_co2" = 0, "max_co2" = 5, "min_n2" = 0, "max_n2" = 0)
+	minimum_survivable_temperature = 270
+	maximum_survivable_temperature = 370
 	unsuitable_atmos_damage = 10
 	unsuitable_heat_damage = 15
-	footstep_type = FOOTSTEP_MOB_SHOE
 	faction = list(FACTION_CLOWN)
-	var/attack_reagent
+	ai_controller = /datum/ai_controller/basic_controller/clown
+	speed = 1.4 //roughly close to simpleanimal clowns
+	///list of stuff we drop on death
+	var/list/loot = list(/obj/effect/mob_spawn/corpse/human/clown)
+	///blackboard emote list
+	var/list/emotes = list(
+		BB_EMOTE_SAY = list("HONK", "Honk!", "Welcome to clown planet!"),
+		BB_EMOTE_HEAR = list("honks", "squeaks"),
+		BB_EMOTE_SOUND = list('sound/items/bikehorn.ogg'), //WE LOVE TO PARTY
+		BB_EMOTE_CHANCE = 5,
+	)
+	///do we waddle (honk)
+	var/waddles = TRUE
 
-/mob/living/simple_animal/hostile/retaliate/clown/Initialize(mapload)
+/mob/living/basic/clown/Initialize(mapload)
 	. = ..()
-	if(attack_reagent)
-		var/static/list/injection_range
-		if(!injection_range)
-			injection_range = string_numbers_list(list(1, 5))
-		AddElement(/datum/element/venomous, attack_reagent, injection_range)
-
-/mob/living/simple_animal/hostile/retaliate/clown/attack_hand(mob/living/carbon/human/user, list/modifiers)
-	..()
-	playsound(loc, 'sound/items/bikehorn.ogg', 50, TRUE)
-
-/mob/living/simple_animal/hostile/retaliate/clown/AttackingTarget(atom/attacked_target)
-	if(!istype(attacked_target, /obj/item/food/grown/banana/bunch))
+	AddElement(/datum/element/footstep, footstep_type = FOOTSTEP_MOB_SHOE)
+	AddComponent(/datum/component/ai_retaliate_advanced, CALLBACK(src, PROC_REF(retaliate_callback)))
+	ai_controller.set_blackboard_key(BB_BASIC_MOB_SPEAK_LINES, emotes)
+	//im not putting dynamic humans or whatever its called here because this is the base path of nonhuman clownstrosities
+	if(waddles)
+		AddElement(/datum/element/waddling)
+	if(LAZYLEN(loot))
```

Comment:
why lazylen if this isn't a lazylist (the default case isn't `null`, it's an initialized list)? just do `length()` instead to avoid confusion